app/models/terminal_executor.rb

```diff
@@ -41,23 +41,24 @@ def execute(*commands, timeout: @timeout)
       script_as_executable(script(commands)) do |command|
         output, _, pid = PTY.spawn(whitelisted_env, command, in: '/dev/null', unsetenv_others: true)
         record_pid(pid) do
-          begin
-            Timeout.timeout(timeout) do
-              stream from: output, to: @output
```

Comment:
How did this work, ie. why was cancel affecting this? wouldn't this keep going until the stream was closed?

Reply:
`cancel` raises and the `raise` interrupted the "read from the stream" code ... so output still came in, but was not read